quickstart/troubleshooting.md

```diff
@@ -0,0 +1,353 @@
+```
+{% endtab %}
+{% endtabs %}
+
+## Coredumps
+
+Coredump is a snapshot of process' memory with auxiliary information
+(PID, state of registers, etc.) saved to a file. It is used for post-mortem
+analysis and it is generated by the operating system in case of a severe error
+(i.e. memory corruption). Using a coredump for a problem analysis requires
+deep knowledge of program internals and is usually done only by developers.
+However, there is a very useful piece of information that even users can
+retrieve and this information alone can often identify the root cause of
+the problem. It is the stack (backtrace). Put differently, the thing that the
+program was doing at the time when it crashed. Here we describe how to get it.
+The steps apply to Ubuntu OS. However different Linux distros would use
+similar steps.
+
+We rely on systemd-coredump that saves and manages coredumps on the system,
+`coredumpctl` utility that is part of the same package and finally
+the `gdb` debugger.
+
+{% tabs %}
+{% tab title="Install systemd-coredump and gdb" %}
+```sh
+sudo apt-get install -y systemd-coredump gdb
+```
+{% endtab %}
+{% endtabs %}
+
+If installed correctly then the global core pattern will be set so that all
+generated coredumps will be piped to the `systemd-coredump` binary.
+
+{% tabs %}
+{% tab title="Verify coredump configuration" %}
+```sh
+cat /proc/sys/kernel/core_pattern
+```
+{% endtab %}
+{% tab title="Output example" %}
+```
+|/lib/systemd/systemd-coredump %P %u %g %s %t 9223372036854775808 %h
+```
+{% endtab %}
+{% endtabs %}
+
+{% tabs %}
+{% tab title="List coredumps" %}
+```sh
+coredumpctl list
+```
+{% endtab %}
+{% tab title="Output example" %}
+```
+TIME                            PID   UID   GID SIG COREFILE  EXE
+Tue 2021-03-09 17:43:46 UTC  206366     0     0   6 present   /bin/mayastor
+```
+{% endtab %}
+{% endtabs %}
+
+If there is a new coredump from mayastor container, the coredump alone won't
+be that useful. GDB needs to access the binary of crashed process in order
+to be able to print at least some information in the backtrace. For that, we
+need to copy the contents of the container's filesystem to the host.
+
+{% tabs %}
+{% tab title="Get ID of the mayastor container" %}
+```sh
+docker ps | grep mayadata/mayastor
+```
+{% endtab %}
+{% tab title="Output example" %}
+```
+b3db4615d5e1        mayadata/mayastor                          "sleep 100000"           27 minutes ago      Up 27 minutes                           k8s_mayastor_mayastor-n682s_mayastor_51d26ee0-1a96-44c7-85ba-6e50767cd5ce_0
+d72afea5bcc2        mayadata/mayastor-csi                      "/bin/mayastor-csi -…"   7 hours ago         Up 7 hours                              k8s_mayastor-csi_mayastor-csi-xrmxx_mayastor_d24017f2-5268-44a0-9fcd-84a593d7acb2_0
+```
+{% endtab %}
+{% endtabs %}
+
+{% tabs %}
+{% tab title="Copy container's filesystem" %}
+```sh
+docker cp b3db4615d5e1:/ /tmp/rootdir
```

Comment:
Should probably fine tune the docker cp to what's strictly necessary as it's copying literally everything including the hosts proc.

Reply:
bin/mayastor and /nix seems to be enough

Reply:
I will address this comment in following PR 👍